provd/proto/user.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package user;
+
+option go_package = "github.com/canonical/ubuntu-desktop-provision/provd/proto";
+
+service UserService {
+  rpc GetUser(GetUserRequest) returns (GetUserResponse);
+  rpc CreateUser(CreateUserRequest) returns (Empty);
+  rpc ValidateUsername(ValidateUsernameRequest) returns (ValidateUsernameResponse);
+}
+
+message Empty {}
+
+message GetUserRequest {
+  string user_id = 1;
+}
+
+message User {
+  string realName = 1;
```

Comment:
All field names should be snake case I think?
https://protobuf.dev/programming-guides/style/#message-field-names

Reply:
Ah yes! It should all be snake :smile:

Reply:
Should be, Go internally changes it all to PascalCase so I think the generator maps it to whatever is idiomatic for that language :slightly_smiling_face: